File: src/main/java/cobweb3d/model/location/LocationDirection.java
package cobweb3d.model.location;

import cobweb3d.model.Topology;

public class LocationDirection extends Location {
	public final Direction direction;

	public LocationDirection(Location l, Direction d) {
		super(l.x, l.y, l.z);
		direction = d;
	}

	public LocationDirection(Location l) {
		this(l, Topology.NONE);
	}
}

File: src/main/java/cobweb3d/model/location/LocationDirection.kt
package cobweb3d.model.location

import java.io.Serializable

data class LocationDirection(var location: Location, var direction: Direction) : Serializable {
    constructor(location: Location) : this(location, Direction.NONE)

    companion object {
        private const val serialVersionUID = 2L
    }
}
